runtime/artifact.go

```diff
@@ -126,11 +127,11 @@ func (context *TaskContext) createArtifact(name string, req []byte) ([]byte, err
 func putArtifact(urlStr, mime string, stream ioext.ReadSeekCloser, additionalArtifacts map[string]string) error {
 	u, err := url.Parse(urlStr)
 	if err != nil {
-		return err
+		panic(errors.Wrap(err, "failed to parse URL"))
```

Comment:
That makes more sense in a panic context.

Reply:
What do you mean?